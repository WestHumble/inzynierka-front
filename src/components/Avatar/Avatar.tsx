import {Image} from "react-native";
import React, {useState} from "react";
import userImg from "../../../assets/images/user-prime.png";

const API_ADDRESS = process.env.EXPO_PUBLIC_API_URL;

const Avatar = ({
  image,
  userId,
  style,
  noCache,
  noCacheId,
}) => {
  const [error, setError] = useState(false);
  var sourceImage = error && !image ? userImg :  { uri: image ?? (`${API_ADDRESS}/api/avatar/get/${userId}` + (noCache ? '?' + noCacheId : '')) }

  return (
      <Image
          source={sourceImage}
          onError={() => setError(true)}
          style={style}

      />
  );
};

export default Avatar;
