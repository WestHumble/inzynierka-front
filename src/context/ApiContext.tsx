import React, {createContext, useState} from 'react';
import axios from 'axios';
import {useNavigation} from "@react-navigation/native";
export const ApiContext = createContext(null);

const API_ADDRESS = process.env.EXPO_PUBLIC_API_URL;
export const ApiProvider = ({children}) => {
    const [userToken, setUserToken] = useState<string>();
    const [userRefreshToken, setUserRefreshToken] = useState<string>();
    let axiosAuthInstance, axiosNoAuthInstance

    const deleteInstances = ()=>{
        axiosAuthInstance = null
        axiosNoAuthInstance = null
    }
    async function getInstance(noAuth, successCallback, errorCallback) {
        if (noAuth) {
            if (!axiosNoAuthInstance) {
                axiosNoAuthInstance = axios.create({
                    baseURL: `${API_ADDRESS}`,
                })
            }
            return axiosNoAuthInstance
        }
        if (!axiosAuthInstance) {
            axiosAuthInstance = axios.create({
                baseURL: `${API_ADDRESS}`,
                headers: {
                    Authorization: `Bearer ${userToken}`,
                }
            })
            axiosAuthInstance.interceptors.response.use(
                response => {
                    return response;
                },
                error => {
                    if (error.request && error.request.status !== 401) {
                        throw error;
                    }
                    let failedRequest = error.config;
                    axios.create({
                        baseURL: `${API_ADDRESS}`,
                    }).post('api/token/refresh', {
                        refresh_token: userRefreshToken
                    }).then((res)=>{
                        setUserToken(res.data.token)
                        setUserRefreshToken(res.data.refresh_token)
                        axiosAuthInstance.defaults.headers['Authorization'] = 'Bearer ' + res.data.token;
                        failedRequest.headers['Authorization'] = 'Bearer ' + res.data.token;

                        return axios.request(failedRequest)
                            .then(res => {
                                if (!res || null === successCallback){
                                    return
                                }
                                successCallback(res)
                            })
                            .catch(res => {
                                if (!res || null === errorCallback){
                                    return
                                }
                                errorCallback(res)
                            })
                    }).catch((res)=>{
                        console.error("Unable to refresh token")
                        if (!res || null === errorCallback){
                            return
                        }
                        errorCallback(res)
                    })
                },
            );
        }
        return axiosAuthInstance
    }
    const post = async (endpoint, body? = null, success = null, error = null, noAuth = false)=> {
        if (!noAuth && !userToken){
            return
        }
        (await getInstance(noAuth, success, error)).post(`/api/${endpoint}`, body).then(res => {
            if (!res || null === success){
                return
            }
            success(res)
        }).catch(res => {
            if (!res || null === error){
                return
            }
            error(res)
        })
    }
    const get = async (endpoint, params? = null, success = null, error = null, noAuth = false)=> {
        if (!noAuth && !userToken){
            return
        }
        (await getInstance(noAuth, success, error)).get(`/api/${endpoint}`, {
            params: params
        }).then(res => {
            if (!res || null === success){
                return
            }
            success(res)
        }).catch(res => {
            if (!res || null === error){
                return
            }
            error(res)
        })
    }
    return (
        <ApiContext.Provider value={{post, get, deleteInstances, userToken, setUserToken, setUserRefreshToken}}>
            {children}
        </ApiContext.Provider>
    );
}