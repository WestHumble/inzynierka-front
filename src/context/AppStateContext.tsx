import React, {createContext, useContext, useEffect, useRef, useState} from 'react';
import axios from 'axios';
import {useNavigation} from "@react-navigation/native";
import {AppState} from "react-native";
import {EventsContext} from "./EventsContext";
import {FriendsContext} from "./FriendsContext";
export const AppContext = createContext(null);

export const AppProvider = ({children}) => {
    const {getPendingEventInvites} = useContext(EventsContext);
    const {getFriendList, getPendingFriendInvites} = useContext(FriendsContext);
    const appState = useRef(AppState.currentState);

    useEffect(() => {
        const subscription = AppState.addEventListener('change', nextAppState => {
            if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
                getPendingEventInvites()
                getFriendList()
                getPendingFriendInvites()
            }
            appState.current = nextAppState;
        });

        return () => {
            subscription.remove();
        };
    }, []);
    return (
        <AppContext.Provider value={{}}>
            {children}
        </AppContext.Provider>
    );
}