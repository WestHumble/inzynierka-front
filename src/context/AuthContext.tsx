import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';
import React, {createContext, useEffect, useState} from 'react';
import { login as loginPost } from "../components/Api/login";


export const AuthContext = createContext(null);

export const AuthProvider = ({children}) => {
    const [isLoading, setIsLoading] = useState(false);
    const [userToken, setUserToken] = useState<string>();

    const login = (email: string, password: string) => {
        setIsLoading(true)
        loginPost(email, password).then(res => {
            setUserToken(res.data.token)
            AsyncStorage.setItem('userToken', res.data.token) 
            setIsLoading(false)
        }).catch(res => {
            console.error(res) 
            setIsLoading(false)
        })
        // axios.post(`http://localhost:81/api/login`, {
        //     email,
        //     password
        // }).then(res => {
        //     setUserToken(res.data.token)
        //     AsyncStorage.setItem('userToken', res.data.token) 
        //     setIsLoading(false)
        // }).catch(res => {
        //     console.error(res) 
        //     setIsLoading(false)
        // })
    }

    const logout = () => {
        setIsLoading(true)
        setUserToken(null)
        AsyncStorage.removeItem('userToken')
        setIsLoading(false)
    }

    const isLoggedIn = async() => {
        try {
            setIsLoading(true)
            let userToken = await AsyncStorage.getItem('userToken')
            setUserToken(userToken == 'null' ? null : userToken)
            setIsLoading(false)
        } catch (e){
            console.log(`isLoggedIn error ${e}`)
        }

    }

    useEffect(() => {
        isLoggedIn()
    }, [])

    return (
        <AuthContext.Provider value={{login, logout, isLoading, userToken}}>
            {children}
        </AuthContext.Provider>
    );
}