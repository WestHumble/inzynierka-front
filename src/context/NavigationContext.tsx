import React, { createContext } from 'react';

export const NavigationContext = createContext(null);
export const NavigationProvider = ({children}) => {
    const navigationRef = React.createRef();

    return (
        <NavigationContext.Provider value={{navigationRef}}>
            {children}
        </NavigationContext.Provider>
    );
}