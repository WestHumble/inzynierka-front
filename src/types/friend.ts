export type Friend = {
    id: number,
    name: string,
    surname: string,
    email: string,
}