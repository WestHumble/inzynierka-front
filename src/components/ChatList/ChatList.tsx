import React, { useContext } from "react";
import {
  View,
  Text,
  SectionList,
  StyleSheet,
  TouchableWithoutFeedback,
  Image,
} from "react-native";
import CustomButton from "../CustomButton";
import { useNavigation } from "@react-navigation/native";
import { ChatContext } from "../../context/ChatContext";
import eventButton from "../../../assets/images/right-alt-yellow.png";

const ChatItem = ({ chat }) => {
  const navigation = useNavigation();
  return (
    <View style={styles.chatContainer}>
      <View style={styles.chatDivText}>
        {chat.is_seen ? (
          ""
        ) : (
          <Text style={styles.nameNewMessage}>Nowa wiadomość</Text>
        )}
        <Text style={styles.name}>{chat.name}</Text>
      </View>
      <View style={styles.chatDivButton}>
        <TouchableWithoutFeedback
          onPress={() => navigation.navigate("Chat", { chatId: chat.id })}
        >
          <Image
            source={eventButton}
            style={styles.eventButtonImg}
            resizeMode="contain"
          />
        </TouchableWithoutFeedback>
        {/* <CustomButton
          text="Otwórz"
          onPress={() => navigation.navigate("Chat", { chatId: chat.id })}
          type="PRIMARY"
          bgColor={undefined}
          fgColor={undefined}
        /> */}
      </View>
    </View>
  );
};

const ChatList = ({ data }) => (
  <SectionList
    sections={data}
    keyExtractor={(item, index) => item.id.toString()}
    renderSectionHeader={({ section: { title } }) => (
      <View style={styles.sectionHeader}>
        <Text style={styles.sectionHeaderText}>{title}</Text>
      </View>
    )}
    renderItem={({ item }) => <ChatItem chat={item} />}
    renderSectionFooter={({ section }) =>
      section.data.length === 0 && (
        <View style={styles.noFriendsContainer}>
          <Text style={styles.noFriendsText}>Brak znajomych</Text>
        </View>
      )
    }
  />
);

const styles = StyleSheet.create({
  chatContainer: {
    padding: 16,
    marginVertical: 4,
    backgroundColor: "#131417",
    borderRadius: 20,
    shadowColor: "black",
    shadowOffset: { width: 0, height: -3 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    flexDirection: "row",
    alignItems: "center",
  },
  name: {
    fontSize: 18,
    fontWeight: "bold",
    color: "#fff",
  },
  sectionHeader: {
    color: "#fff",
    padding: 8,
    marginTop: 12,
  },
  sectionHeaderText: {
    fontSize: 21,
    fontWeight: "bold",
    color: "#fff",
  },
  noFriendsContainer: {
    padding: 16,
    alignItems: "center",
  },
  noFriendsText: {
    fontSize: 16,
    color: "#888",
  },
  chatDivText: {
    flex: 2 / 3,
    flexDirection: "column",
    justifyContent: "space-between",
  },
  chatDivButton: {
    flex: 1 / 3,
    flexDirection: "row",
    justifyContent: "flex-end",
  },
  eventButtonImg: {
    width: 30,
    height: 30,
  },
  nameNewMessage: {
    color: "#ffffff",
    backgroundColor: "#F2B138",
    padding: 5,
    borderRadius: 10,
    marginBottom: 5,
    width: 125,
  },
});

export default ChatList;
