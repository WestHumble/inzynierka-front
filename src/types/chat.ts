import {Friend} from "./friend";

export type Chat = {
    id: number,
    name: string,
    event_name: string,
    participants: Friend[],
    is_seen: boolean,
    messages: ChatMessage[],
    last_message_date: string,
}

export type ChatMessage = {
    id: number,
    chat_id: number,
    sender_id: number,
    message_text: string,
    created_at: string,
    sent_by_logged_user: boolean,
}