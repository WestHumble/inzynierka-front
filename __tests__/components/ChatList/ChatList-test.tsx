import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from '@jest/globals';
import ChatList from "../../../src/components/ChatList";
import {Friend} from "../../../src/types/friend";
import {ChatMessage} from "../../../src/types/chat";
import {NavigationContext} from "../../../src/context/NavigationContext";
import Avatar from "../../../src/components/Avatar";

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigate,
        }),
    };
});
test('empty', () => {
    const tree = renderer.create(
        <ChatList data={[
            {
                title: "Chaty",
                data: [],
            },
        ]}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('one chat', () => {
    const tree = renderer.create(
        <ChatList data={[
            {
                title: "Chaty",
                data: [{
                    id: 1,
                    name: "name",
                    event_name: "event_name",
                    participants: [],
                    is_seen: true,
                    messages: [],
                    last_message_date: '2024-01-01 11:22',
                }],
            },
        ]}/>
    )
    expect(tree).toMatchSnapshot();
});



test('multiple chats', () => {
    const tree = renderer.create(
        <ChatList data={[
            {
                title: "Chaty",
                data: [{
                    id: 1,
                    name: "name",
                    event_name: "event_name",
                    participants: [],
                    is_seen: true,
                    messages: [],
                    last_message_date: '2024-01-01 11:22',
                },{
                    id: 2,
                    name: "name",
                    event_name: "event_name",
                    participants: [],
                    is_seen: true,
                    messages: [],
                    last_message_date: '2024-01-01 11:22',
                },{
                    id: 3,
                    name: "name",
                    event_name: "event_name",
                    participants: [],
                    is_seen: true,
                    messages: [],
                    last_message_date: '2024-01-01 11:22',
                }],
            },
        ]}/>
    )
    expect(tree).toMatchSnapshot();
});