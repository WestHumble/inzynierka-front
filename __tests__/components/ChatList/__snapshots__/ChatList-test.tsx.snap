// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [],
        "title": "Chaty",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    [
      0,
    ]
  }
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
            }
          }
        >
          Chaty
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "padding": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 16,
            }
          }
        >
          Brak znajomych
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`multiple chats 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [
          {
            "event_name": "event_name",
            "id": 1,
            "is_seen": true,
            "last_message_date": "2024-01-01 11:22",
            "messages": [],
            "name": "name",
            "participants": [],
          },
          {
            "event_name": "event_name",
            "id": 2,
            "is_seen": true,
            "last_message_date": "2024-01-01 11:22",
            "messages": [],
            "name": "name",
            "participants": [],
          },
          {
            "event_name": "event_name",
            "id": 3,
            "is_seen": true,
            "last_message_date": "2024-01-01 11:22",
            "messages": [],
            "name": "name",
            "participants": [],
          },
        ],
        "title": "Chaty",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    [
      0,
    ]
  }
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
            }
          }
        >
          Chaty
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          name
           
          seen
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 20,
                "height": 60,
                "justifyContent": "center",
                "marginVertical": 5,
                "padding": 0,
                "width": "100%",
              },
              {
                "backgroundColor": "#F2B138",
              },
              {},
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontWeight": "bold",
                },
                {
                  "color": "white",
                },
                {},
                undefined,
              ]
            }
          >
            Otwórz
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          name
           
          seen
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 20,
                "height": 60,
                "justifyContent": "center",
                "marginVertical": 5,
                "padding": 0,
                "width": "100%",
              },
              {
                "backgroundColor": "#F2B138",
              },
              {},
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontWeight": "bold",
                },
                {
                  "color": "white",
                },
                {},
                undefined,
              ]
            }
          >
            Otwórz
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          name
           
          seen
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 20,
                "height": 60,
                "justifyContent": "center",
                "marginVertical": 5,
                "padding": 0,
                "width": "100%",
              },
              {
                "backgroundColor": "#F2B138",
              },
              {},
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontWeight": "bold",
                },
                {
                  "color": "white",
                },
                {},
                undefined,
              ]
            }
          >
            Otwórz
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;

exports[`one chat 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [
          {
            "event_name": "event_name",
            "id": 1,
            "is_seen": true,
            "last_message_date": "2024-01-01 11:22",
            "messages": [],
            "name": "name",
            "participants": [],
          },
        ],
        "title": "Chaty",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    [
      0,
    ]
  }
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "color": "#fff",
            "marginTop": 12,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 21,
              "fontWeight": "bold",
            }
          }
        >
          Chaty
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#131417",
            "borderRadius": 20,
            "marginVertical": 4,
            "padding": 16,
            "shadowColor": "black",
            "shadowOffset": {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          name
           
          seen
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 20,
                "height": 60,
                "justifyContent": "center",
                "marginVertical": 5,
                "padding": 0,
                "width": "100%",
              },
              {
                "backgroundColor": "#F2B138",
              },
              {},
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontWeight": "bold",
                },
                {
                  "color": "white",
                },
                {},
                undefined,
              ]
            }
          >
            Otwórz
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
